src/minmax.rs

```diff
@@ -0,0 +1,273 @@
+// Copyright (C) 2018-2019, Cloudflare, Inc.
+// Copyright (C) 2017, Google, Inc.
```

Comment:
Where does this code come from?

Reply:
this is from Google open sources code from BSD license.
https://groups.google.com/forum/#!topic/bbr-dev/3RTgkzi5ZD8